### PR TITLE
Photo path improvements and camera fixes

### LINKED ISSUE #256: Absolute path saved when taking pictures
With actual version 0.10.11, there is no more crash whn taking a picture, cool. 

But, the path written in the database is absolute (/storage/...), instead of relative (DCIM/...), even if relative paths are configured in project. This is really annoying, because project is no more compatible between mobile and desktop. Could it be fixed ? Thanks !

### PR DESCRIPTION
**Pull request by signedav**
1. Photos are now named in the format YYYYMMDDHHmmSSsss.jpg and in the relative path DCIM/<imagename> Fix #256 

2. The settings for the native camera are read directly and are not looped over featureforms.

3. Fix of small bug when using native camera that in the background the new camera implementation appeared.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://opengis.ch/download/qfield/ci-builds/qfield-dev-258-970b39f709f15dfa1cf072593849301dae1844bb-armv7.apk) and [x86](https://opengis.ch/download/qfield/ci-builds/qfield-dev-258-970b39f709f15dfa1cf072593849301dae1844bb-x86.apk)

File: src/qml/editorwidgets/ExternalResource.qml [hunk 1]
```diff
@@ -24,22 +24,23 @@ Item {
     fillMode: Image.PreserveAspectFit
 
     source: {
-      if (image.status === Image.Error)
+      if (image.status === Image.Error) {
         Style.getThemeIcon("ic_broken_image_black_24dp")
-      else if (currentValue)
-        if ( featureUseNativeCamera )
+      } else if (currentValue) {
+        if ( settings.value("useNativeCamera", false) )
```
Comment (m-kuhn) [inline src/qml/editorwidgets/ExternalResource.qml]: Is `settings` available here?

Comment (signedav) [reply src/qml/editorwidgets/ExternalResource.qml]: Why not?

Comment (signedav) [reply src/qml/editorwidgets/ExternalResource.qml]: But I guess it shoud be `valueBool` as well here.

Comment (signedav) [reply src/qml/editorwidgets/ExternalResource.qml]: And it's not even needed, since the path is the same anyway...

**Comment by giedrioks26** [comment]
test apk (armv7) works fine. time id writes good. writes to relative path.
[ekranvaizdis_2018-03-30_10-59-25](https://user-images.githubusercontent.com/6436791/38129876-7c912d0c-3409-11e8-81a1-7de9011dab2a.png)

File: src/qml/editorwidgets/ExternalResource.qml (hunk 1 above)
Comment (m-kuhn) [reply src/qml/editorwidgets/ExternalResource.qml]: in this case, kill it here.
and it looks like it works, so all good :+1:

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://opengis.ch/download/qfield/ci-builds/qfield-dev-258-b3bb4fbcc2538484f7a06c234f2e98428f69e3a9-armv7.apk) and [x86](https://opengis.ch/download/qfield/ci-builds/qfield-dev-258-b3bb4fbcc2538484f7a06c234f2e98428f69e3a9-x86.apk)

**Comment by m-kuhn** [comment]
Nice @signedav can you click the "Delete branch" button above?

**Comment by FilipjFry** [comment]
Is there a feature that allows one to name the picture files from a station ID ?

I guess I would have to edit those lines ?

var timestamp = (new Date()).toISOString().replace(/[^0-9]/g, "")
var filename = timestamp+'.jpg';

Fry

**Comment by m-kuhn** [comment]
Hi Fry, good idea.
I would recommend to open [a new issue](https://github.com/opengisch/QField/issues/new) here for your questions, then we'll be able to discuss focussed about it.

**Pull request merged**